lib/formats/archiver.js

```diff
@@ -16,21 +17,62 @@ function compress(filepath, data, archive) {
       }
     });
     archive.on('error', reject);
-    archive.on('finish', resolve);
-
+    archive.on('finish', () => {
+      resolve(response);
+    });
     archive.pipe(output);
 
-    for (const filename in data) {
-      const content = data[filename];
+    for (const [ filename, content ] of Object.entries(data.json || {})) {
+      archive.append(content, { name: filename });
+    }
 
-      if (content.endsWith('/')) {
-        archive.directory(content, filename);
-        continue;
-      }
+    compressAttachments(apos, archive, data.attachments || {})
+      .then((res) => {
+        response = res;
+        archive.finalize();
+      });
+  });
+}
 
-      archive.append(content, { name: filename });
+// If one image download fail should we stop the entire process?
+async function compressAttachments(apos, archive, attachments = {}) {
+  const promises = Object.entries(attachments).map(([ name, url ]) => {
+    return new Promise((resolve, reject) => {
+      apos.http.get(url, { originalResponse: true })
+        .then((res) => {
+          res.body.on('error', reject);
+          res.body.on('end', resolve);
+
+          archive.append(res.body, {
+            name: `attachments/${name}`
+          });
+        })
+        .catch(reject);
+    });
+  });
+
+  const chunkedPromises = chunkPromises(promises, 5);
+  let attachmentError = false;
+
+  for (const chunk of chunkedPromises) {
+    const results = await Promise.allSettled(chunk);
+    for (const { status } of results) {
+      if (status === 'rejected' && !attachmentError) {
+        attachmentError = true;
+      }
     }
+  }
+
+  return { attachmentError };
+}
 
-    archive.finalize();
+function chunkPromises(attachments, max) {
+  const length = Math.ceil(attachments.length / max);
+  return Array(length).fill([]).map((chunk, i) => {
+    const position = i * max;
+    return [
+      ...chunk,
+      ...attachments.slice(position, position + max)
```

Comment:
Yes because of this line:
`const length = Math.ceil(attachments.length / max);`
we create as many arrays as needed chunks. The `ceil` allows to be sure we have a last array to put the last promises even if the don't fill it.
Could be tested indeed.